### DISCUSSION TITLE
Setup with CC2531 is very unstable

### DISCUSSION THREAD

**Discussion by wizake** [Koenkk/zigbee2mqtt#5809] [Devices]
Hey community!

I have a setup consisting of one CC2531 coordinator and three CC2531's as routers. (source routing firmware)
I also have about 25 Livolo switches that I paired and controlled with HA. Using latest version of 'z2m edge' add-on

This setup is very unstable. The LQI of the routers (especially the ones upstairs) varies a lot.. and it drops to 0 very often.
My switches stop responding, and I get all the errors you can think of: from timeouts to MEM_ERROR and a bunch I cant remember.

This forces me to do a full rebuild of Home Assistant every week.

I have ordered a more powerful stick (CC2652R), to try it our, but until it arrives I'm struggling to keep the network up and running. 
Because of the Livolo switches which are locked on channel 26, I cannot change the channel. I suspect that is maybe the issue.

Any suggestions what I can try ?

I couldn't find any detailed information about how to create multiple networks (I hear it's possible) with 2 coordinators maybe (since I have 4 CC2531 sticks).

**Comment by wizake** [comment]
![image](https://user-images.githubusercontent.com/33145382/104930332-b1ac3780-59ad-11eb-8a3a-af147950435a.png)

this is how LQI looks all of the time

**Comment by wizake** [comment]
And this is how my network maps looks as of today, after a clean install of HA, and pairing everything
(excuse the friendly names, they're in Romanian)

![image](https://user-images.githubusercontent.com/33145382/104930666-21babd80-59ae-11eb-98e0-8fa1e07e6fc0.png)

**Reply by potormarius** [reply]
Hey, I am also trying to get the Livolo switches integrated. What do you use as routers?

I am having trouble that after poweroff the switches are not connecting to the coordinator. How did you solve it?

**Comment by DrTron** [comment]
The culprit is most likely the CC2531, it's just not very good, neither as coordinator nor as router.
I now use two CC2530+CC2591 routers and link quality has improved significantly. All of my devices spread over the whole house have 60-90.

**Reply by wizake** [reply]
You could be right.
Just got a new coordinator today (CC2652R stick) and I got rid of all CC2531's. With the new coordinator and 2 IKEA signal repeaters i get a better link quality. About 90-100 on all devices.

If this setup proves to be stable, then we might have a winner :)

**Reply by avlisiur78** [reply]
Hi, have the set up proves to be more stable? 
Do you have any issues, with the Livolo devices that are connected to the coordinator and to the router at the same time? In my network if this happens the Livolo device stop working, and get alive again if connected only to one of the two (coordinator or router).

**Reply by wizake** [reply]
yes, it was much more stable.
not perfect though. I have to leave the coordinator in pairing mode (i have an automation that turns pairing mode on because the addon puts a timer and turn it off after approx 5 minutes) otherwise it happens that some switches lose connection to the routers/coordinator and I cant control them

This also happens wven when pairing mode is on, but much more rare
